src/System.ComponentModel.TypeConverter/tests/Drawing/ImageConverterTests.cs

```diff
@@ -0,0 +1,354 @@
+            Assert.True(_imgConv.CanConvertTo(null, _imageStr.GetType()), "_imageStr.GetType()");
+            Assert.True(_imgConv.CanConvertTo(typeof(byte[])), "byte[] (no context)");
+            Assert.True(_imgConv.CanConvertTo(null, typeof(byte[])), "byte[]");
+            Assert.True(_imgConv.CanConvertTo(null, _imageBytes.GetType()), "_imageBytes.GetType()");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(Rectangle)), "Rectangle");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(Point)), "Point");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(PointF)), "PointF");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(Size)), "Size");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(SizeF)), "SizeF");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(object)), "object");
+            Assert.False(_imgConv.CanConvertTo(null, typeof(int)), "int");
+
+            Assert.True(_imgConvFrmTD.CanConvertTo(typeof(string)), "TD string (no context)");
+            Assert.True(_imgConvFrmTD.CanConvertTo(null, typeof(string)), "TD string");
+            Assert.True(_imgConvFrmTD.CanConvertTo(null, _imageStr.GetType()), "TD _imageStr.GetType()");
+            Assert.True(_imgConvFrmTD.CanConvertTo(typeof(byte[])), "TD byte[] (no context)");
+            Assert.True(_imgConvFrmTD.CanConvertTo(null, typeof(byte[])), "TD byte[]");
+            Assert.True(_imgConvFrmTD.CanConvertTo(null, _imageBytes.GetType()), "TD _imageBytes.GetType()");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(Rectangle)), "TD Rectangle");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(Point)), "TD Point");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(PointF)), "TD PointF");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(Size)), "TD Size");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(SizeF)), "TD SizeF");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(object)), "TD object");
+            Assert.False(_imgConvFrmTD.CanConvertTo(null, typeof(int)), "TD int");
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void ConvertFrom()
+        {
+            Image newImage = (Image)_imgConv.ConvertFrom(null, CultureInfo.InvariantCulture, _imageBytes);
+
+            Assert.Equal(_image.Height, newImage.Height);
+            Assert.Equal(_image.Width, newImage.Width);
+
+            Assert.Equal("(none)", _imgConvFrmTD.ConvertTo(null, CultureInfo.InvariantCulture, null, typeof(string)));
+
+            newImage = (Image)_imgConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, _imageBytes);
+
+            Assert.Equal(_image.Height, newImage.Height);
+            Assert.Equal(_image.Width, newImage.Width);
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void ConvertFrom_BadString()
```

Comment:
Might be worth unifying all this NotSupportedException tests into one that is called CanConvertFrom_ThrowsNotSupportedException and have all the asserts in one test.